core/src/main/kotlin/org/evomaster/core/output/naming/RestActionTestCaseNamingStrategy.kt

```diff
@@ -161,14 +181,52 @@ open class RestActionTestCaseNamingStrategy(
                 result.add(and)
             }
         }
+
+        val numberQueryParams = getNegativeNumberQueryParams(queryParams)
+        numberQueryParams.forEachIndexed { index, queryParam ->
+            result.add("negative")
+            result.add(queryParam.name)
+            if (index != numberQueryParams.lastIndex) {
+                result.add(and)
+            }
+        }
+
+        val emptyStringQueryParams = getEmptyStringQueryParams(queryParams)
+        emptyStringQueryParams.forEachIndexed { index, queryParam ->
+            result.add("empty")
+            result.add(queryParam.name)
+            if (index != emptyStringQueryParams.lastIndex) {
+                result.add(and)
+            }
+        }
     }
 
+    // TODO: need to check if the BooleanGene is not enclosed in an OptionalGene
     private fun getBooleanQueryParams(queryParams: List<QueryParam>): List<QueryParam> {
-        return queryParams.filter { it.getGeneForQuery() is BooleanGene && (it.getGeneForQuery() as BooleanGene).value }
+        return queryParams.filter {
+            val wrappedGene = getWrappedGene(it)
+            wrappedGene is BooleanGene && wrappedGene.value
+        }
+    }
+
+    // TODO: need to check if the NumberGene is not enclosed in an OptionalGene
+    private fun getNegativeNumberQueryParams(queryParams: List<QueryParam>): List<QueryParam> {
+        return queryParams.filter {
+            val wrappedGene = getWrappedGene(it)
+            wrappedGene is NumberGene<*> && wrappedGene.value.toLong() < 0
+        }
+    }
+
+    // TODO: need to check if the StringGene is not enclosed in an OptionalGene
+    private fun getEmptyStringQueryParams(queryParams: List<QueryParam>): List<QueryParam> {
+        return queryParams.filter {
+            val wrappedGene = getWrappedGene(it)
+            wrappedGene is StringGene && wrappedGene.getValueAsRawString().trim().isEmpty()
+        }
     }
 
-    private fun removeSolvedDuplicates(duplicatedIndividuals: MutableSet<EvaluatedIndividual<*>>, disambiguatedIndividuals: Set<EvaluatedIndividual<*>>) {
-        duplicatedIndividuals.removeAll(disambiguatedIndividuals)
+    private fun getWrappedGene(queryParam: QueryParam): Gene {
+        return (queryParam.getGeneForQuery() as OptionalGene).gene
```

Comment:
there is an already `Gene.getWrappedGene`. use that. the situation can be even more complex, including null and choice genes